dao.py

```diff
@@ -645,6 +665,17 @@ def check_creds_and_get_session_id_or_none(self, username, password):
         else:
             return None
 
+    def check_creds(self, username, password):
```

Comment:
Oh looking a little closer it might be because it finds no user in the db and then returns None...